compiler/incremental-compilation-impl/test/org/jetbrains/kotlin/incremental/classpathDiff/ClasspathSnapshotTestCommon.kt

```diff
@@ -126,11 +128,19 @@ abstract class ClasspathSnapshotTestCommon {
             }
 
             val classesDir = tmpDir.newFolder()
-            org.jetbrains.kotlin.test.MockLibraryUtil.compileKotlin(
+            // Note: Calling the following is simpler:
+            //     org.jetbrains.kotlin.test.MockLibraryUtil.compileKotlin(
+            //         srcDir.path, classesDir, extraClasspath = classpath.map { it.path }.toTypedArray())
+            // However, it currently fails with UnsupportedClassVersionError, so we have to launch a new kotlinc process instead
+            // (Linux only).
```

Comment:
This code is only temporary, ideally we should use `K2JVMCompiler` directly and compile Kotlin files on the fly instead of checking in the class files.

Right now it is not possible (as I described in the comments), so this is just a quick hack for me for compile the Kotlin files locally (it's not run on the server).

I can make this work on Windows if you think it's important?

Reply:
It is ok to check it in as-is, but pls add an explicit check that throws if the platform is unsupported.

Reply:
Done - I've added support for Windows.